sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.vb

```diff
@@ -21,9 +21,9 @@ Namespace Tests.TestCases
 
         Private Sub DirectAddHandlers()
             'Inline version
-            'Secondary@+1
-            CreateRQ().ServerCertificateValidationCallback = Function(sender, certificate, chain, SslPolicyErrors) True         'Noncompliant  {{Enable server certificate validation on this SSL/TLS connection}}
-            '          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                         ^^^^
+            CreateRQ().ServerCertificateValidationCallback = Function(sender, certificate, chain, SslPolicyErrors) True
+            '          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ {{Enable server certificate validation on this SSL/TLS connection}}
+            '                                                                                                      ^^^^ Secondary@-1
```

Comment:
I've added typo situation like this:
```
        static bool Sec(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)        {
            return true;    //Secondary [x1, x2]
        }
        void Check()        {
            CreateRQ().ServerCertificateValidationCallback += Sec;    //Noncompliant [x2]
            CreateRQ().ServerCertificateValidationCallback += Sec;    //Noncompliant [x2]
        }
```
and I got `Unexpected secondary issue [x2] on line 14 with message ''.` error message. 
I would expect [x1] to be unexpected in the message. Aren't there swapped variables somewhere?

Reply:
Actually the message is OK here. Since there is twice [x2] primary raising the issue, there is an additional secondary issue on that line with id [x2] that was unexpected.
As mentioned in the other comment, it should not be allowed to have 2 primary with same id anyway.
But in the following case:
```
        static bool Sec(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)        {
            return true;    //Secondary [x1, x2, x3]
        }
        void Check()        {
            CreateRQ().ServerCertificateValidationCallback += Sec;    //Noncompliant [x2]
            CreateRQ().ServerCertificateValidationCallback += Sec;    //Noncompliant [x3]
        }
```
then we would have a message for expected but not raised issue. I just checked and I ve improved that message so that it now mentions the type and id of the expected issue that was not raised.